### ISSUE TITLE
Bootstrapping the package loader

### ISSUE THREAD

**Issue by deepakjois** [deepakjois/luatex-harfbuzz#8]
One of the problems with using LuaRocks for developing LuaTeX related packages is that the default package loader in LuaTeX uses kpse. kpse overrides the standard Lua package loading behavior that LuaRocks relies on.

It is possible to get around it by [overriding the LuaTeX package loader](http://tex.stackexchange.com/questions/218855/using-lualatex-and-sqlite3/219228#219228), to revert it to the standard Lua behavior. We already have that functionality in [ufy](https://github.com/deepakjois/ufy). But `ufy` itself is a LuaRocks package and needs to be loaded before anything else, and hence we have a bootstrapping problem when we invoke LuaTeX.

Currently, we get around it by using LuaTeX’s `--lua` flag to load a pre-init file that executes before LuaTeX is fully initialized (and hence retains the default package loading behavior), and loads ufy in the global environment. Then we invoke the relevant functions in the TeX file, once LuaTeX is fully initialized. This method however relies on the user invoking LuaTeX with the `--lua` flag and having a pre-init file located somewhere.

A more elegant solution to this problem might be to create a CTAN module (maybe something called `luapackageloader`) that can encapsulate the package loading functionality. Assuming this CTAN module is properly installed inside the TeX distribution, we can invoke it as:

```tex
\input luapackageloader

% Now it should be possible to load a LuaRocks package
\directlua{require("luabidi")}
```
and then continue loading the rocks.

**Comment by deepakjois** [comment]
Created [luapackageloader](https://github.com/deepakjois/luapackageloader). Need to add some docs and upload to CTAN.

**Comment by deepakjois** [comment]
Package uploaded to CTAN. Now we wait…

**Comment by deepakjois** [comment]
Ok, so the package has been approved and it’s already been a while so I am not sure if all the mirrors have picked it up: https://www.ctan.org/pkg/luapackageloader

I am not sure how to install this from CTAN, do I use `tlmgr install luapackageloader`?

Can somebody verify that they can install this package from CTAN? @michal-h21, @khaledhosny?

**Comment by khaledhosny** [comment]
The TeX Live team syncs the packages from CTAN, after that you can use `tlmgr` to install it. But since TL 2016 is frozen the package will not appear there, in TL 2017 is also temporary frozen for the release so the package will appear sometime after the final release.

There is no way to install from CTAN directly other than downloading and installing the package manually.

**Comment by deepakjois** [comment]
I installed the package “manually”, i.e. I unzipped the contents of the zip in the right path and then ran `texhash`. That seems to work for now.

I am going to go ahead and update the sources to use this. The package will get picked up eventually I guess. I will also update the README to specify TexLive 2017 as a dependency, because that is probably the version of LuaTeX that would work with luatex-harfbuzz anyway.
